### PR TITLE
434 fix scms security bugs

### PR DESCRIPTION
**Pull request by paulbourelly999**
# PR Details
## Description

Removed/consolidated unused or duplicate configuration parameters. Updated DSRC Signing logic to log HTTP error code and message with error responses from SCMS container
## Related Issue
#434 

## Motivation and Context
Fix security implemetation

## How Has This Been Tested?
Tested with Cohda RSU and OBU by @jwilliammartin 

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
[V2XHUB Contributing Guide](https://github.com/usdot-fhwa-OPS/V2X-Hub/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/v2i-hub/DsrcImmediateForwardPlugin/src/DsrcMessageManagerPlugin.cpp [hunk 1]
```diff
@@ -390,7 +393,7 @@ void DsrcMessageManagerPlugin::SendMessageToRadio(IvpMessage *msg)
 			else
 				os << "Priority=7" << "\n" << "TxMode=CONT" << "\n" << "TxChannel=" << _messageConfigMap[configIndex].Channel << "\n";
 			os << "TxInterval=0" << "\n" << "DeliveryStart=\n" << "DeliveryStop=\n";
-			os << "Signature="<< _signature << "\n" << "Encryption=False\n";
+			os << "Signature= "<< (signState == 1 ? "True" : "False") << "\n" << "Encryption=False\n";
```
Comment (jwillmartin) [inline src/v2i-hub/DsrcImmediateForwardPlugin/src/DsrcMessageManagerPlugin.cpp:396]: Need to remove this space to be "Signature=" instead of "Signature= "

Comment (paulbourelly999) [reply src/v2i-hub/DsrcImmediateForwardPlugin/src/DsrcMessageManagerPlugin.cpp:396]: Good catch

Comment (paulbourelly999) [reply src/v2i-hub/DsrcImmediateForwardPlugin/src/DsrcMessageManagerPlugin.cpp]: Resolved!

**Review by jwillmartin — APPROVED** [review]
All else looks good to me.

File: src/v2i-hub/MessageReceiverPlugin/src/MessageReceiverPlugin.cpp [hunk 2]
```diff
@@ -502,11 +459,23 @@ int MessageReceiverPlugin::Main()
 						{
 							result+=buffer; 
 						}
-					} catch (...) {
+					} catch (std::exception const & ex) {
+					
 						pclose(pipemsg); 
-						throw; 
+						SetStatus<uint>(Key_SkippedSignVerifyError, ++_skippedSignVerifyErrorResponse);
```
Comment (dan-du-car) [inline src/v2i-hub/MessageReceiverPlugin/src/MessageReceiverPlugin.cpp:465]: Add inline comment about what this SetStatus does, and what parameter has passed in

File: src/v2i-hub/MessageReceiverPlugin/src/MessageReceiverPlugin.cpp [hunk 3]
```diff
@@ -502,11 +459,23 @@ int MessageReceiverPlugin::Main()
...
+						PLOG(logERROR) << "Error parsing Messages: " << ex.what();
+						continue;
+; 
 					}
+					PLOG(logDEBUG1) << "SCMS Contain response = " << result << std::endl;
 					cJSON *root   = cJSON_Parse(result.c_str());
+					cJSON *status = cJSON_GetObjectItem(root, "code");
+					if ( status ) {
+						cJSON *message = cJSON_GetObjectItem(root, "message");
+						SetStatus<uint>(Key_SkippedSignVerifyError, ++_skippedSignVerifyErrorResponse);
```
Comment (dan-du-car) [inline src/v2i-hub/MessageReceiverPlugin/src/MessageReceiverPlugin.cpp:475]: The same as the above

File: src/v2i-hub/DsrcImmediateForwardPlugin/src/DsrcMessageManagerPlugin.cpp [hunk 4]
```diff
@@ -44,22 +45,10 @@ DsrcMessageManagerPlugin::DsrcMessageManagerPlugin(std::string name) : PluginCli
 
 	_muteDsrc = false;
 	SetSystemConfigValue("MuteDsrcRadio", _muteDsrc, false);
-	UpdateConfigSettings();
-
-	// @SONAR_STOP@
-
-
-	GetConfigValue<string>("HSMurl",baseurl);
```
Comment (dan-du-car) [inline src/v2i-hub/DsrcImmediateForwardPlugin/src/DsrcMessageManagerPlugin.cpp:52]: Why is this removed?

File: docker/Dockerfile-depl [hunk 5]
```diff
@@ -3,7 +3,8 @@ FROM ubuntu:focal-20220113 AS install_dependencies
 ENV DEBIAN_FRONTEND=noninteractive
 RUN apt-get update  && apt-get install -y cmake git build-essential libgtest-dev libssl-dev qtbase5-dev \
     zip libmysqlcppconn-dev libboost1.71-all-dev libmysqlclient-dev uuid-dev zlib1g libxerces-c-dev qtbase5-dev \
-    libcurl4-openssl-dev libgps-dev libsnmp-dev librdkafka-dev libjsoncpp-dev libev-dev libuv1-dev libcpprest-dev
+    libcurl4-openssl-dev libgps-dev libsnmp-dev librdkafka-dev libjsoncpp-dev libev-dev libuv1-dev libcpprest-dev \
+    curl
```
Comment (dan-du-car) [inline docker/Dockerfile-depl:7]: Is this curl dependency needed for this change?

Comment (rob-baily) [reply docker/Dockerfile-depl]: @dan-du-car  and @paulbourelly999 
With the latest changes that Cody and I have merged we have removed the separate Dockerfile-depl.  I have merged them into this branch to ensure there is no merge conflict and so this change is overcome by events.  It did not seem that curl was needed but if it is we can put it in the correct place.

File: src/v2i-hub/DsrcImmediateForwardPlugin/src/DsrcMessageManagerPlugin.cpp (hunk 4 above)
Comment (paulbourelly999) [reply src/v2i-hub/DsrcImmediateForwardPlugin/src/DsrcMessageManagerPlugin.cpp:52]: Duplicate configuration parameter

File: src/v2i-hub/MessageReceiverPlugin/src/MessageReceiverPlugin.cpp (hunk 2 above)
Comment (paulbourelly999) [reply src/v2i-hub/MessageReceiverPlugin/src/MessageReceiverPlugin.cpp:465]: Will do

File: src/v2i-hub/MessageReceiverPlugin/src/MessageReceiverPlugin.cpp (hunk 3 above)
Comment (paulbourelly999) [reply src/v2i-hub/MessageReceiverPlugin/src/MessageReceiverPlugin.cpp:475]: Will do

File: docker/Dockerfile-depl (hunk 5 above)
Comment (paulbourelly999) [reply docker/Dockerfile-depl]: Yes curl is needed for the calls in the ImmediateForward and MessageReceiver plugins. When you say you removed the Dockerfile-depl does that mean you replaced the existing one with the Dockerfile-depl. I created this docker file to remove unused dependencies if I remember correctly.

Comment (rob-baily) [reply docker/Dockerfile-depl]: @paulbourelly999 The process for Docker has been modified.  Previously the Dockerfile was used in the Sonar Build and for x64 buils while the Dockerfile-depl was used for only arm builds.  Cody and I figured out that we did not need Sonar Scanner to use a Docker image with everything in it so we have moved some of the commands that were in the Dockerfiles (there was a  lot of duplication) into scripts as these scripts can also be used for developers working directly on a VM.   Either curl can be added in the install_dependencies.sh script or this can be reworked to use libcurl like is done in the CARMA Cloud plugin.  Probably using the library is better than making a system call.
